### PR TITLE
Milokang/menu component(feedback only)

### PR DESCRIPTION
**Pull request by milocosmopolitan**
**Please note that test spec files are not updated according to the recent changes.

Sorry for a lot of file changes to review, I feel like I should've make pull request on this earlier, with each components, but all scaffolded components are part of header menu.
Each groups of components are modularized to help developers to write less code for dependency injection of each spec file.

New modules are added
- MenuModule
- ListModule
- IconModule

HeaderMenu is generated as object `/app/components/header/header-menu.model.ts`
HeaderAction and HeaderType is removed from header component and handled by MenuService

Directives are added for logout or close toggle, this eliminates unit testing on button click to check if certain functions are invoked. Now we can just make a isolated unit testing for each directives and leave button click testing to e2e feature testing.

### CODE REVIEW

File: src/app/components/header/header-menu.model.ts [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+  container: 'navbar-user',
+  menu: {
+    id: 'user',
+    name: 'header',
+  },
+  items: [
+    {
+      icon: 'fa fa-pencil-square-o',
+      text: 'Edit Profile',
+      link: '#/app/extra/profile'
```
Comment (cmelion) [inline src/app/components/header/header-menu.model.ts]: we are using the html5 history api, right?
These links have #s

**Comment by cmelion** [comment]
```
ERROR in /software/ariel/capdash2-client/src/app/app.module.ts (26,28): Cannot find module './components/icon/icon.module'.

ERROR in Error encountered resolving symbol values statically. Could not resolve ./components/icon/icon.module relative to /software/ariel/capdash2-client/src/app/app.module.ts., resolving symbol AppModule in /software/ariel/capdash2-client/src/app/app.module.ts, resolving symbol AppModule in /software/ariel/capdash2-client/src/app/app.module.ts

ERROR in ./src/app/app.module.ts
Module not found: Error: Can't resolve './components/icon/icon.module' in '/software/ariel/capdash2-client/src/app'
 @ ./src/app/app.module.ts 30:0-59
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 webpack/hot/dev-server ./src/main.ts
```

File: src/app/app.module.ts [hunk 2]
```diff
@@ -8,18 +8,22 @@ import {NgReduxRouterModule} from '@angular-redux/router';
...
-import {HeaderComponent} from './components/header/header.component';
-import {HomeComponent} from './containers/home/home.component';
-import {HotelsModule} from './containers/hotels/hotels.module';
-import {ItemModule} from './core/ajax/item/item.module';
-import {LoginComponent} from './containers/login/login.component';
-import {OfflineUnitsModule} from './containers/offline-units/offline-units.module';
-import {PageNotFoundComponent} from './core/page-not-found/page-not-found.component';
-import {SidebarComponent} from './components/sidebar/sidebar.component';
-import {StoreModule} from './store/store.module';
+import { ItemModule } from './core/ajax/item/item.module';
```
Comment (milocosmopolitan) [inline src/app/app.module.ts]: Just installed LineSorter to my webstorm :)

File: src/app/components/header/header-menu.model.ts (hunk 1 above)
Comment (milocosmopolitan) [inline src/app/components/header/header-menu.model.ts]: I was just placing the same value from what header.component.html had, but I did removed hash from the url string

**Comment by cmelion** [comment]
I am seeing "unsafe:javascript" when hovering over the menus, the links do not function either.
Also it would be nice if the hover used the active css.  Please check the original capdash app and see if they have existing behavior.

**Comment by milocosmopolitan** [comment]
@cfulnecky I’ve made changes due to your reviews. Whole Icon directory just wasn't included while I was running git add . command. Although .gitignore doesn't have any condition for icon directory, I wasn't able to see icon directory when I executed git status, so I had to add it with --force. @graigjanssen  just made sure it is running with hmr and json-server code after last change made to PR.

**Comment by milocosmopolitan** [comment]
Pipe for url sanitization has been implemented & missing css has been added. However I think  further investigation needs to be made regarding bootstrap version diff from [DSSCapApp](https://github.com/arielpartners/DSSCapApp)

screen shot 2017-06-05 at 2 07 30 pm
.text-black in included in style.css which seems like complied version of bootstrap

screen shot 2017-06-05 at 2 11 51 pm
Bootstrap version in DSSCapApp is 3.3.5

screen shot 2017-06-05 at 2 10 51 pm
Bootstrap version in capdash2-client is 3.3.7
- this doesn't have css properties assigned to .text-black

**Comment by cmelion** [comment]
I'm not concerned about the actual class used to achieve the desired effect, only that the desired effect is achieved.

**Comment by milocosmopolitan** [comment]
I see. Is there any easy way or tools you use to compare UI ready version on the browser and current working repo at the same time or perhaps different branch of repo? I was trying to go back and forth between master branch and working branch to check the UI difference.

**Comment by milocosmopolitan** [comment]
Thanks for pointing that out, I will look into it

**Pull request merged**